### ISSUE TITLE
Make MauiDoneAccessoryView DoneButton accessible

### ISSUE THREAD

**Issue by lucacivale** [dotnet/maui#32825]
### Description

PR #24645 added a Done button to numeric entry's keyboard.

Currently it's not accessible because no Id is assigned.
IMHO Maui should set some Id(Microsoft.Maui.Platform.MauiDoneAccessory.DoneButton?) to enable UI-Tests

### Steps to Reproduce

1. Add a Entry with numeric keyboard to a page
2. Inspect elements

### Version with bug

10.0.10

### Is this a regression from previous behavior?

No, this is something new

### Affected platforms

iOS

### Relevant log output

```shell

```

**Comment by jfversluis** [comment]
@lucacivale would you be able to test the artifacts from the linked PR and let uw know if that works for you?

**Comment by lucacivale** [comment]
@jfversluis had a look at it but there is an (strange?) issue.
Appium doesn't find the Button at all and even Apple's Accessibility Inspector has trouble finding it.

Apple's inspector does not show the Done button inside the Element hierarchy but if select the Done button directly to inspect its properties I can see the set `Microsoft.Maui.Platform.MauiDoneAccessory.DoneButton` Id

I even tried setting all accessibility properties I could but it didn't make any difference 

```
doneButton.AccessibilityIdentifier = DoneButtonAutomationId;
doneButton.AccessibilityLabel = DoneButtonAutomationId;
doneButton.IsAccessibilityElement = true;
doneButton.AccessibilityTraits = UIAccessibilityTrait.Button;
doneButton.AccessibilityAttributedHint = new("TestHint");
doneButton.AccessibilityFrame = new CGRect(0, 0, 100, 44);
doneButton.AccessibilityValue = "TestValue";
doneButton.AccessibilityActivate();
```

Even tried to check the actual UIView to see if the properties are set. It also looked good

```
DispatchQueue.MainQueue.DispatchAsync(() =>
{
    var view = button.ValueForKey(new NSString("view")) as UIView;
});
```

**Comment by Ahamed-Ali** [comment]
This issue has been validated in Visual Studio Code 1.106.2 with MAUI versions 10.0.0, 10.0.10, and 10.0.11. It cannot be reproduced on the iOS platform.

The Done accessibility ID is displayed correctly in the Appium Inspector.

<img width="800" height="521" alt="Image" src="https://github.com/user-attachments/assets/349c5e27-5d56-4901-8b2b-444836ba1661" />

**Note**: I have also confirmed that the Done ID is correctly recognized in the UI test.

<img width="2344" height="908" alt="Image" src="https://github.com/user-attachments/assets/bf29eadb-8421-4fef-aa0f-2b4a0f47dc41" />

[MauiApp16.zip](https://github.com/user-attachments/files/23742699/MauiApp16.zip)

**Comment by lucacivale** [comment]
@Ahamed-Ali can you try this using iOS 26 built with Xcode 26?

**Comment by Ahamed-Ali** [comment]
I have checked on `iOS 26`, and instead of showing the “`Done`” text, it displays an `icon` representing `Done`. In iOS 26.0, this icon button has the automation ID “`selected`,” and it also works correctly in the UI test.

<img width="2196" height="1396" alt="Image" src="https://github.com/user-attachments/assets/d19d0f30-ff38-47d4-903d-17f583808aca" />

<img width="800" height="267" alt="Image" src="https://github.com/user-attachments/assets/387387b3-7d71-49a5-ba95-1b327c191b36" />

**Comment by lucacivale** [comment]
@Ahamed-Ali thank you, really appreciate the effort.

I think I found the issue.
If you present the page as a sheet the done button isn't found anymore.

```
<ContentPage ...
    ios:Page.ModalPresentationStyle="PageSheet"
    xmlns:ios="clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.iOSSpecific;assembly=Microsoft.Maui.Controls">
    <Entry
        HeightRequest="100"
        Keyboard="Numeric"
        VerticalOptions="Start" />
</ContentPage>
```

**Comment by Ahamed-Ali** [comment]
I have checked the accessibility ID of the Done button after `presenting the page as a sheet using ModalPresentationStyle`. The ID is still shown correctly on both iOS 18 and iOS 26.

### 18.0 iOS
<img width="800" height="538" alt="Image" src="https://github.com/user-attachments/assets/59f17093-ec1f-401f-9395-e8a1317d982b" />

### 26.0 iOS
<img width="2656" height="1758" alt="Image" src="https://github.com/user-attachments/assets/9fa4358a-24f8-4295-883b-22e77c3071de" />

Could you please share the repro project you used to replicate the issue, along with screenshots? This will help me investigate the issue further.

**Comment by lucacivale** [comment]
@Ahamed-Ali sorry my bad.
I can confirm that the return button is discovered by Appium and the new Id `Microsoft.Maui.Platform.MauiDoneAccessory.DoneButton` is assigned.

Seems like Apple's Accessibility Inspector has some issues as it does not discover all views in different scenarios.

@Ahamed-Ali sorry again for any inconvenience and thanks for the quick help!
